src/http/ngx_http_wasm_api.c

```diff
@@ -762,30 +782,63 @@ ngx_http_wasm_req_get_header(ngx_http_request_t *r, char *key,  int32_t key_size
         key_buf = (u_char *) key;
     }
 
-    for (i = 0; /* void */ ; i++) {
+    if (key_size > 0 && key_buf[0] == ':') {
+        for (i = 0; i < WASM_H2_HEADER_STATIC_TABLE_ENTRIES; i++) {
+            wh = &wasm_h2_header_static_table[i];
 
-        if (i >= part->nelts) {
-            if (part->next == NULL) {
-                break;
+            if ((size_t) key_size != wh->name.len) {
+                continue;
             }
 
-            part = part->next;
-            header = part->elts;
-            i = 0;
-        }
+            if (ngx_strncasecmp(key_buf, wh->name.data, wh->name.len) == 0) {
 
-        if (header[i].hash == 0) {
-            continue;
-        }
+                switch (wh->ty) {
+                case WASM_H2_HEADER_PATH:
+                    val = r->unparsed_uri.data;
+                    val_len = r->unparsed_uri.len;
+                    break;
 
-        if ((size_t) key_size != header[i].key.len) {
-            continue;
+                case WASM_H2_HEADER_METHOD:
+                    val = r->method_name.data;
+                    val_len = r->method_name.len;
+                    break;
+
+                /* todo: scheme https://github.com/api7/wasm-nginx-module/issues/47 */
+                default:
+                    break;
+                }
+
+                if (val) break;
+            }
         }
+    }
+
+    if (val == NULL) {
```

Comment:
We should use an early return instead of checking the state of a variable.

Reply:
please check whether the processing method after the update is correct.